static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js

```diff
@@ -166,6 +166,28 @@ const registerIframeListener = (iframeId: string) => {
     });
 };
 
+const addServerSideRenderingTestParameterToLink = (rawUrl: string): string => {
+    const serverSideRenderingField = 'ssr';
+    const randomNumber = Math.random();
+    if (
+        randomNumber === undefined ||
```

Comment:
Can randomNumber ever be `undefined`? IIRC `Math.random()` will always return a number.

Reply:
It is theoretically possible, see [this stackoverflow answer](https://stackoverflow.com/questions/29072744/math-random-javascript-function-undefined-on-chrome)

I thought this at first but @johnduffell  said he'd been stung by this before, so put it in for added peace of mind

Reply:
after someone sent me the list of ways that undefined and null and anything else can be coerced into something weird, I just felt paranoid!

Reply:
Interesting! That makes sense. 

We could be more specific by checking `typeof randomNumber === 'number'`